app/models/event.rb

```diff
@@ -177,6 +177,83 @@ def email_addresses_of_type_without_notification_sent(type)
     applications.collect { |a| a.user.email }
   end
 
+  # Returns a string of all email addresses of accepted applications
+  #
+  # @param none
+  # @return [String] Concatenation of all email addresses of accepted applications, seperated by ','
+  def email_addresses_of_accepted_applicants
+    participants.collect { |user| user.email }.join(',')
+  end
+
+  # Returns a new email to a set of participants
+  #
+  # @param all Boolean If set to true, addresses all participants
+  # @param groups Array<Integer> The group-ids whoose members should be addressed
+  # @param users Array<Integer> The user-ids which should be addressed
+  # @return Email new email
+  def generate_participants_email(all, groups, users)
+    Email.new(
+        :hide_recipients => false,
+        :recipients => email_addresses_of_participants(all, groups, users),
+        :reply_to => '',
+        :subject => '',
+        :content => ''
+    )
+  end
+
+  # Returns all email addresses of user that fit the given criteria
+  #
+  # @param all Boolean If set to true, return addresses of all participants
+  # @param groups Array<Integer> The group-ids whoose members-addresses should be looked up
+  # @param users Array<Integer> The user-ids whoose addresses should be returned
+  # @return String list of email addresses
+  def email_addresses_of_participants(all, groups, users)
+    if all
+      email_addresses_of_accepted_applicants
+    else
+      (email_addresses_of_groups(groups) + email_addresses_of_users(users)).uniq.join(',')
+    end
+  end
+
+  # Returns a list of tuples containing all participant names and their id
+  #
+  # @return Array<Array<String, Int>>
+  def participants_with_id
+    participants.map{|participant| [participant.profile.name, participant.id]}
+  end
+
+  # Returns a list of group names and their id
+  #
+  # @return Array<Array<String, Int>>
+  def groups_with_id
+    existing = ParticipantGroup::GROUPS.select do |group_id,_|
+      group_id != 0 and participant_groups.where(:group => group_id).any?
```

Comment:
The style guide suggests always using `&&` instead of `and` because of weird operator precedence.